### PR TITLE
Events page

### PR DESCRIPTION
**Pull request by johnrich85**
Changes proposed in this pull request:

- Adds the events page

### CODE REVIEW

File: site/en/meet-the-team/events/an-event-apart/index.md
Comment (rachelandrew) [inline site/en/meet-the-team/events/an-event-apart/index.md:2]:
```suggestion
title: An Event Apart
```

File: site/en/meet-the-team/events/an-event-apart/index.md
Comment (rachelandrew) [inline site/en/meet-the-team/events/an-event-apart/index.md:5]:
```suggestion
  digital design, UX, content, code, and more.
```

File: site/en/meet-the-team/events/an-event-apart/index.md
Comment (rachelandrew) [inline site/en/meet-the-team/events/an-event-apart/index.md:8]:
```suggestion
date: '2022-10-10'
```

File: site/en/meet-the-team/events/devfest-nantes/index.md
Comment (rachelandrew) [inline site/en/meet-the-team/events/devfest-nantes/index.md:4]:
```suggestion
  The Devfest (Developers Festival), is a technical conference destined to
```

File: site/en/meet-the-team/events/w3c-developer-meetup/index.md
Comment (rachelandrew) [inline site/en/meet-the-team/events/w3c-developer-meetup/index.md:5]:
```suggestion
  of demonstrations, presentations, and discussions with the W3C community.
```

File: site/en/meet-the-team/events/w3c-developer-meetup/index.md
Comment (rachelandrew) [inline site/en/meet-the-team/events/w3c-developer-meetup/index.md:15]:
```suggestion
    description: 'For the first time ever, all major browser vendors, and other stakeholders, have come together to solve the top browsers compatibility issues identified by web developers. Interop 2022 will improve the experience of developing for the web in 15 key areas. In this talk, you’ll find out how we got here, what the project focuses on, how success will be measured, and how you can track progress.'
```

**Review by rachelandrew — COMMENTED** [review]
The date for An Event Apart was wrong, corrected it and a few missing bits of punctuation etc.

File: site/en/meet-the-team/index.njk
Comment (devnook) [inline site/en/meet-the-team/index.njk:2]: We try to avoid having .njk files in language directories, because this cannot be picked up by our translation service. Is there a reason to not have the code directly in events.njk instead of includes here?

**Comment by devnook** [comment]
What will be entry point to the /meet-the-team? I don't see it in navigation

File: site/_js/utils/truncate-string.js
Comment (devnook) [inline site/_js/utils/truncate-string.js:26]: Please add test

File: site/_js/web-components/enhanced-event-card.js
Comment (devnook) [inline site/_js/web-components/enhanced-event-card.js:19]: Please add jsdoc

File: site/_js/web-components/truncate-text.js
Comment (devnook) [inline site/_js/web-components/truncate-text.js:41]: Please use i18n filter to make this string translatable

File: site/_js/web-components/truncate-text.js
Comment (devnook) [inline site/_js/web-components/truncate-text.js:20]: Please add jsdoc

File: site/_js/web-components/truncate-text.js
Comment (devnook) [inline site/_js/web-components/truncate-text.js:43]: Please remove the listeners in cleanup too.

**Comment by devnook** [comment]
Events tab do not align - the blue underline is too short in Past events:

Screenshot 2022-09-27 at 02 20 53
Screenshot 2022-09-27 at 02 20 12

File: site/_js/web-components/truncate-text.js
Comment (devnook) [inline site/_js/web-components/truncate-text.js:53]: Why is typescript complaining here?

File: site/_js/web-components/truncate-text.js
Comment (devnook) [inline site/_js/web-components/truncate-text.js:2]: Nit: 2022

File: site/_scss/blocks/_event-card.scss
Comment (devnook) [inline site/_scss/blocks/_event-card.scss:4]: let's use booleans for logic values?

File: site/_shortcodes/EventCard.js
Comment (devnook) [inline site/_shortcodes/EventCard.js:71]: Please use i18n filter for translated content

File: site/_shortcodes/EventCard.js
Comment (devnook) [inline site/_shortcodes/EventCard.js:155]: i18n

File: site/_shortcodes/EventCard.js
Comment (devnook) [inline site/_shortcodes/EventCard.js:171]: i18n

File: site/en/meet-the-team/events/events.11tydata.js
Comment (devnook) [inline site/en/meet-the-team/events/events.11tydata.js:4]: Can you explaing what canLinkToList do? I see it is used in algolia, but I don't get the rationale behind it.

**Review by devnook — CHANGES_REQUESTED** [review]
Thanks @johnrich85 ! I left the first round of comments

File: site/en/meet-the-team/events/events.11tydata.js
Comment (johnrich85) [reply site/en/meet-the-team/events/events.11tydata.js:4]: As it stands currently, only items with a permalink are added to the index. We're not adding a permalink for events, but do want them to be included in the search results. The newly introduced variables, canLinkToList and listUrl allow this to happen.

File: site/_js/web-components/truncate-text.js
Comment (johnrich85) [reply site/_js/web-components/truncate-text.js:53]: Although it would never happen, it was complaining that fullText is of type string|undefined, rather than string. To fix that I've declared the properties upfront in the constructor.

**Comment by johnrich85** [comment]
> Events tab do not align - the blue underline is too short in Past events:
> 
> Screenshot 2022-09-27 at 02 20 53
> 
> Screenshot 2022-09-27 at 02 20 12

Are you referring to the horizontal space between the two titles @devnook ? If so I believe that's the intended design.


> What will be entry point to the /meet-the-team? I don't see it in navigation

I'm not sure actually, so i've passed the query over to Niamh.

File: site/en/meet-the-team/index.njk
Comment (johnrich85) [reply site/en/meet-the-team/index.njk]: No problem - have switched to an MD file and moved the includes to the layout. Just FYI I asked about i18n and apparently the events are going up in English only. Understand it's best to future-proof regardless, of course!

File: site/_js/utils/truncate-string.js
Comment (johnrich85) [reply site/_js/utils/truncate-string.js]: Thanks @devnook . I've added tests for this.

File: site/_js/web-components/truncate-text.js
Comment (johnrich85) [reply site/_js/web-components/truncate-text.js:50]: Would it be ok to delay this one until the next phase @devnook ? Importing the  i18n filter on the frontend results in an error as it relies on the fs module (ultimately were going to have the same issue with the EventCard shortcode as that'll need to work on the frontend too). Not sure i have time to look into a workaround for that given that we're supposed to be releasing either today or tomorrow.

Comment (devnook) [reply site/_js/web-components/truncate-text.js:50]: Ok. Please create an issue to track it and link to this PR as context.

File: site/_scss/blocks/_event-card.scss
Comment (devnook) [inline site/_scss/blocks/_event-card.scss:4]: Still not boolean?

File: site/_shortcodes/EventCard.js
Comment (devnook) [inline site/_shortcodes/EventCard.js:1]: Please add License and jsdoc

File: site/_shortcodes/EventCard.js
Comment (devnook) [inline site/_shortcodes/EventCard.js:103]: This should live in i18n.events not common - or is it reused somewhere? Same for other i18n strings

File: site/_scss/blocks/_event-card.scss
Comment (johnrich85) [reply site/_scss/blocks/_event-card.scss:4]: I think this is the standard for boolean attributes (as defined here https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes). Can switch out to use 'true' for true and '' for false though if that's the preference?

File: site/en/meet-the-team/events/events.11tydata.js
Comment (devnook) [reply site/en/meet-the-team/events/events.11tydata.js:4]: but what kind of List does it refer to? and what is listUrl?

File: site/_collections/algolia.js
Comment (devnook) [inline site/_collections/algolia.js:65]: LEt's use some more generic name for this property. We have VirtualCollectionItem type to describe item that do now have a corresponsing .md file and therefore permalink. Maybe .isVirtualItem = true to make this indexing change more generic?

File: site/_collections/algolia.js
Comment (devnook) [inline site/_collections/algolia.js:79]: it feels like this logic should live in the events page rather than here. maybe events should have a property called algoliaUrl that overwrites item.url if present? Then the individual page/item itself can decide how it wants to be indexed in algolia, including redirects.

**Review by devnook — CHANGES_REQUESTED** [review]
Reviewed

File: site/_collections/algolia.js
Comment (johnrich85) [reply site/_collections/algolia.js]: Makes sense! Just pushing up the changes now. 

Also, It's just occurred to me that since we've introduced tabs to the page, the anchors are only going to work for items in the first tab. Ill try to address that today but wondering if we need a contingency so that we don't block the release. If we're happy with the events not showing in search initially, we could always just set isVirtualItem to false for now.

File: site/_scss/blocks/_event-card.scss
Comment (devnook) [reply site/_scss/blocks/_event-card.scss:4]: I think the meaning is that 'false' does not resolve to False in attributes, but is used as any other string. The boolean way would be to add and remove show-details attribute (without value) rather than changing it's string value. But it's ok either way, it's just a nit - can stay as is.

File: site/en/meet-the-team/events/devfest-nantes/index.md
Comment (devnook) [inline site/en/meet-the-team/events/devfest-nantes/index.md:10]: This needs to be automatically generated in an 11tydata file because authors will not know how to add this value.

File: site/en/meet-the-team/events/devfest-nantes/index.md
Comment (devnook) [inline site/en/meet-the-team/events/devfest-nantes/index.md:21]: Pls remove empty line from frontmatter - will break some of our parsers

File: site/_collections/algolia.js
Comment (devnook) [reply site/_collections/algolia.js]: Let's move the algolia logic from this PR and please submit an issue to cover it in next release, so that we can launch on time. Wdyt?

**Review by devnook — APPROVED** [review]
I approve to enable timely launch, but please remvoe Algolia logic before merging, as it is still faulty in places.

**Pull request merged**
